packages/core/useMediaControls/index.ts

```diff
@@ -369,6 +369,8 @@ export function useMediaControls(target: MaybeRef<HTMLMediaElement | null | unde
       source.setAttribute('src', src)
       source.setAttribute('type', type || '')
 
+      useEventListener(source, 'error', sourceErrorEvent.trigger)
```

Comment:
I don't think this will be automatically cleaned up since it is nested inside of a watchEffect and effectScope isn't supported yet, so I think we will need to manually clean this up like we did starting on line 476. It might be cleaner if rename `listeners` to `disposables` and push each listener into the array of disposables instead of manually assigning them like I previously did.

Reply:
@wheatjs Since the `sources` is never reactive, it doesn't seem to make sense to use `useEventListener`. Maybe should use the built-in `addEventListener`?

```js
source.addEventListener('error', sourceErrorEvent.trigger)
```

And if I'm not mistaken, the listener of events on the DOM node will be automatically deleted together with this DOM node.

https://github.com/vueuse/vueuse/blob/59a47a1e7f7929383cf5ba7fc76055caf36b356c/packages/core/useMediaControls/index.ts#L363

Reply:
If they will be removed when the element is unmounted then `addEventListener` is fine